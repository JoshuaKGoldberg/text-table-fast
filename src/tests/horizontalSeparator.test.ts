import { describe, expect, test } from "vitest";

import { textTable } from "../index.js";

describe("align", () => {
	test.each<[string[][], string]>([
		[[["abc", "12"]], ""],
		[[["abc", "12"]], " "],
		[[["abc", "12"]], "  "],
		[[["abc", "12"]], "   "],
		[[["abc", "12"]], "\t"],
		[[["abc", "12"]], "|"],
		[
			[
				["abc", "12"],
				["def", "3456"],
			],
			"|",
		],
		[
			[
				["abc", "12"],
				["def", "3456"],
			],
			" | ",
		],
	])("%j %j", (rows, horizontalSeparator) => {
		expect(
			"\n" + textTable(rows, { horizontalSeparator }) + "\n",
		).toMatchSnapshot();
	});
});
