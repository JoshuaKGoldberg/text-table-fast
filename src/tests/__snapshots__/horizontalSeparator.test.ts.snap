// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`align > [["abc","12"],["def","3456"]] " | " 1`] = `
"
abc | 12
def | 3456
"
`;

exports[`align > [["abc","12"],["def","3456"]] "|" 1`] = `
"
abc|12
def|3456
"
`;

exports[`align > [["abc","12"]] "   " 1`] = `
"
abc   12
"
`;

exports[`align > [["abc","12"]] "  " 1`] = `
"
abc  12
"
`;

exports[`align > [["abc","12"]] " " 1`] = `
"
abc 12
"
`;

exports[`align > [["abc","12"]] "" 1`] = `
"
abc12
"
`;

exports[`align > [["abc","12"]] "\\t" 1`] = `
"
abc	12
"
`;

exports[`align > [["abc","12"]] "|" 1`] = `
"
abc|12
"
`;
